### PR TITLE
DOCS-#3690: Add Slack and paper link to README

### PR DESCRIPTION
**Pull request by dorisjlee**
Signed-off-by: Doris Lee <dorisjunglinlee@gmail.com>

## What do these changes do?

Addresses parts of #3690. Changes to `More information` section of README by adding Slack and paper link.

- [x] commit message follows format outlined [here](https://modin.readthedocs.io/en/latest/contributing.html)
- [ ] passes `flake8 modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [ ] passes `black --check modin/ asv_bench/benchmarks scripts/doc_checker.py`
- [x] signed commit with `git commit -s` 
- [x] Resolves #? 
- [ ] tests added and passing
- [ ] module layout described at `docs/developer/architecture.rst` is up-to-date

### CODE REVIEW

**Review by RehanSD — DISMISSED** [review]
Looks great to me! I'm re-running tests and am happy to merge once they're green!

File: README.md [hunk 1]
```diff
@@ -190,8 +191,10 @@ more information, and checkout [the difference between Modin and Dask!](https://
...
-- [Documentation](https://modin.readthedocs.io/en/latest/)
-- Ask questions or participate in discussions on our [Discourse](https://discuss.modin.org)
-- Join our mailing list [modin-dev@googlegroups.com](https://groups.google.com/forum/#!forum/modin-dev)
-- Submit bug reports to our [GitHub Issues Page](https://github.com/modin-project/modin/issues)
-- Contributions are welcome! Open a [pull request](https://github.com/modin-project/modin/pulls)
+- Read the [documentation](https://modin.readthedocs.io/en/latest/) for more information.
+- Check out [our paper](http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf) to learn more about how Modin works.
+- Ask questions or participate in discussions on our [Discourse](https://discuss.modin.org).
+- Let us know how you're using Modin! Join our community [Slack](https://modin.org/slack.html) to discuss and ask questions.
+- Join our mailing list: [modin-dev@googlegroups.com](https://groups.google.com/forum/#!forum/modin-dev).
```
Comment (devin-petersohn) [inline README.md:198]:
```suggestion
```

File: README.md [hunk 2]
```diff
@@ -190,8 +191,10 @@ more information, and checkout [the difference between Modin and Dask!](https://
 
 ### More information and Getting Involved
 
-- [Documentation](https://modin.readthedocs.io/en/latest/)
-- Ask questions or participate in discussions on our [Discourse](https://discuss.modin.org)
-- Join our mailing list [modin-dev@googlegroups.com](https://groups.google.com/forum/#!forum/modin-dev)
-- Submit bug reports to our [GitHub Issues Page](https://github.com/modin-project/modin/issues)
-- Contributions are welcome! Open a [pull request](https://github.com/modin-project/modin/pulls)
+- Read the [documentation](https://modin.readthedocs.io/en/latest/) for more information.
+- Check out [our paper](http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf) to learn more about how Modin works.
```
Comment (devin-petersohn) [inline README.md:195]:
```suggestion
- Check out [our paper](http://www.vldb.org/pvldb/vol13/p2033-petersohn.pdf) to learn more about the theory underlying Modin.
```

**Review by devin-petersohn — COMMENTED** [review]
Thanks @dorisjlee, left a couple of nits.

**Comment by dorisjlee** [comment]
Updated based on suggestions, thanks @devin-petersohn!

**Review by vnlitvinov — APPROVED** [review]
LGTM!

**Review by devin-petersohn — APPROVED** [review]
LGTM, thanks @dorisjlee!

**Pull request merged**